### PR TITLE
Patch updates

### PR DESCRIPTION
**Pull request by gregsdennis**
Was going to do both patch and logic here, but decided not to.

### CODE REVIEW

File: JsonPatch/PatchOperation.cs [hunk 1]
```diff
@@ -159,7 +159,7 @@ private class Model
 
 	public override PatchOperation Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
 	{
-		var model = JsonSerializer.Deserialize<Model>(ref reader, options)!;
+		var model = JsonSerializer.Deserialize(ref reader, PatchSerializerContext.Default.Model)!;
```
Comment (gregsdennis) [inline JsonPatch/PatchOperation.cs:162]: Not really sure why, but this isn't deserializing the contents.  I get an empty model instance.

Comment (jevansaks) [reply JsonPatch/PatchOperation.cs:162]: Does Model have a JsonConverter? If not, take a look at the generated code for PatchSerializerContext and see what it produces. I looked at this and the serializer it generated for Model was just empty. Maybe there are some attributes you can add to Model that help the JsonSerializable code generator know how to serialize the fields?

Comment (gregsdennis) [reply JsonPatch/PatchOperation.cs:162]: I would expect the serializer to be able to handle POCOs out of the box.  Strange.  I'll try adding a custom converter and see if it makes a difference.

Comment (jevansaks) [reply JsonPatch/PatchOperation.cs:162]: Is there a behavior difference in net8 versus netcoreapp3.1 for the tests? Could be a bug in STJ 6, I've seen a bunch of issues like that. I'll build your branch locally and look too.

Comment (jevansaks) [reply JsonPatch/PatchOperation.cs:162]: Also I agree, the documentation says that it should be able to generate the serializers for primitive types.

Comment (jevansaks) [reply JsonPatch/PatchOperation.cs:162]: This is weird. I'm debugging "PatchExtensionTests.Add" test and there's a bunch of Model deserializes that work fine and then the one that the test passes in doesn't deserialize. I haven't figured it out yet.

Comment (jevansaks) [reply JsonPatch/PatchOperation.cs:162]: Looks like the Model's properties are "Op" and the test is providing a payload where the properties are "op". I suspect Options.PropertyNameCaseInsensitive is coming into play here but I haven't finished tracking it down.

Comment (gregsdennis) [reply JsonPatch/PatchOperation.cs:162]: I bet you're right: case sensitivity.  I can mark with `[JsonProperty()]`.  That will likely fix it.  Good thinking.

File: JsonPatch/JsonPatch.cs [hunk 2]
```diff
@@ -15,6 +16,14 @@ namespace Json.Patch;
 [JsonConverter(typeof(PatchJsonConverter))]
 public class JsonPatch : IEquatable<JsonPatch>
 {
+#if NET8_0_OR_GREATER
+	/// <summary>
+	/// A TypeInfoResolver that can be used for serializing <see cref="JsonPointer"/> objects. Add to your custom
+	/// JsonSerializerOptions's TypeInfoResolver or TypeInfoResolveChain.
+	/// </summary>
+	public static IJsonTypeInfoResolver JsonTypeResolver => PatchSerializerContext.Default;
```
Comment (jevansaks) [inline JsonPatch/JsonPatch.cs:24]: I called this TypeInfoResolver in JsonSchema. I don't have a strong preference, but we should be consistent.

File: JsonPatch/JsonPatch.cs [hunk 3]
```diff
@@ -95,7 +104,7 @@ public class PatchJsonConverter : JsonConverter<JsonPatch>
 	/// <returns>The converted value.</returns>
 	public override JsonPatch Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
 	{
-		var operations = JsonSerializer.Deserialize<List<PatchOperation>>(ref reader, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase })!;
```
Comment (jevansaks) [inline JsonPatch/JsonPatch.cs:98]: This is where we lost the CamelCase (and presumably the case insensitivity) from before. You can add it back through the JsonSourceGeneratorOptions attribute on the PatchSerializerContext if that makes sense? But I also saw you did name aliases on the Model objects, that might be cleaner.

**Review by jevansaks — APPROVED** [review]
Looks good, left a couple suggestions.

**Pull request merged**